### DISCUSSION TITLE
goroutines

### DISCUSSION THREAD

**Discussion by itsabgr** [vlang/v#18610] [Ideas]
did you see [libcsp](https://libcsp.dev/)?
goroutines for c;
I think it's a good starting point to add goroutine support in V

**Comment by enghitalo** [comment]
This look performative for now.
![image](https://user-images.githubusercontent.com/63821277/200309054-cb419249-9b5a-41fc-a133-f81d700a6763.png)
![image](https://user-images.githubusercontent.com/63821277/200309275-3594c034-0877-4ec2-95c2-13cfa0d8e5a7.png)

**Comment by bogen85** [comment]
Co-routines use to be part of posix see https://stackoverflow.com/questions/15014647/why-was-ucontext-added-to-and-then-removed-from-posix , but were removed (deprecated).

I looked briefly at libscp. At a low level it looks a bit like ucontext.

**Comment by itsabgr** [comment]
I patched the V compiler to use libcsp but it requires compiling with GCC
